vamb/__main__.py

```diff
@@ -195,6 +197,45 @@ def __init__(
         self.slr = slr
 
 
+class TaxonomyOptions:
+    __slots__ = ["taxonomy_path", "taxonomy_predictions_path", "no_predictor"]
+
+    def __init__(
+        self,
+        taxonomy_path: Optional[Path],
+        taxonomy_predictions_path: Optional[Path],
+        no_predictor: bool,
+    ):
+        assert isinstance(taxonomy_path, (Path, type(None)))
+        assert isinstance(taxonomy_predictions_path, (Path, type(None)))
+
+        if taxonomy_path is None and taxonomy_predictions_path is None and no_predictor:
+            raise argparse.ArgumentTypeError(
+                "You must specify either taxonomy_path or taxonomy_predictions_path or run without --no_predictor flag"
+            )
```

Comment:
If there is no taxonomy, no results of a predictor, but also no predictor to run, then there is no taxonomy to include, and this whole set of parameters is meaningless.

However, if `not no_predictor`, than the `taxonomy_path` must exist otherwise there is nothing to predict from. I'll make sure there is a check on that